micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -177,16 +177,29 @@ public void bindTo(MeterRegistry registry) {
         counter(registry, "hibernate.statements", "The number of prepared statements that were released",
             Statistics::getCloseStatementCount, "status", "closed");
 
-        // Second Level Caching
-        Arrays.stream(statistics.getSecondLevelCacheRegionNames())
-                .forEach(regionName ->  {
+        // Second Level Caching:
+        long numRegions = Arrays.stream(statistics.getSecondLevelCacheRegionNames())
+                .filter(regionName -> hasDomainDataRegionStatistics(regionName))
+                .map(regionName -> {
                     counter(registry, "hibernate.second.level.cache.requests", "The number of cacheable entities/collections successfully retrieved from the cache",
-                            stats -> stats.getDomainDataRegionStatistics(regionName).getHitCount(), "region", regionName, "result", "hit");
+                            stats -> stats.getSecondLevelCacheStatistics(regionName).getHitCount(), "region", regionName, "result", "hit");
                     counter(registry, "hibernate.second.level.cache.requests", "The number of cacheable entities/collections not found in the cache and loaded from the database",
-                            stats -> stats.getDomainDataRegionStatistics(regionName).getMissCount(), "region", regionName, "result", "miss");
+                            stats -> stats.getSecondLevelCacheStatistics(regionName).getMissCount(), "region", regionName, "result", "miss");
                     counter(registry, "hibernate.second.level.cache.puts", "The number of cacheable entities/collections put in the cache",
-                            stats -> stats.getDomainDataRegionStatistics(regionName).getPutCount(), "region", regionName);
-                });
+                            stats -> stats.getSecondLevelCacheStatistics(regionName).getPutCount(), "region", regionName);
+                    return regionName;
+                })
+                .count();
+
+        if ( numRegions <= 0 ) {
```

Comment:
Should not this be `==`?

Reply:
should be. I guess old-school paranoia. ;)

Reply:
I know what you are talking about. :D